### PR TITLE
Specific Lengths blacked and ruffed

### PR DESCRIPTION
**Pull request by BadPrograms**

### CODE REVIEW

File: src/LineageTree/lineageTree.py [hunk 1]
```diff
@@ -2037,18 +2038,40 @@ def get_simple_tree(self, r: int, time_resolution: int = 1) -> tuple:
...
+            if the cell lifetime ends before the time_end it returns the whole cycle.
+
+            Args:
+                cell (int): Id of the cell
+                time_end (int): The final timepoint
+
+            Returns:
+                (int): Length of the life cycle of a cell up to the time_end timepoint.
+            """
+            if self.time[cell] >time_end:
```
Comment (leoguignard) [inline src/LineageTree/lineageTree.py:2052]: You sure you ran black on that file?!

```python
self.time[cell] >time_end
```
should be
```python
self.time[cell] > time_end
```

File: src/LineageTree/lineageTree.py [hunk 2]
```diff
@@ -2018,7 +2018,8 @@ def get_ancestor_at_t(self, n: int, time: int = None):
             ancestor = self.predecessor.get(ancestor, [-1])[0]
         return ancestor
 
-    def get_simple_tree(self, r: int, time_resolution: int = 1) -> tuple:
+    
+    def get_simple_tree(self, r: int, time: int = 500, time_resolution: int = 1) -> tuple:
```
Comment (leoguignard) [inline src/LineageTree/lineageTree.py:2022]: The `time` parameter is not explicit enough.

I am not sure what it is you'd like to do here tbh.

Comment (leoguignard) [reply src/LineageTree/lineageTree.py:2022]: Also you should add `time` description in the docstring

File: src/LineageTree/lineageTree.py [hunk 3]
```diff
@@ -2037,18 +2038,40 @@ def get_simple_tree(self, r: int, time_resolution: int = 1) -> tuple:
...
+
         if not hasattr(self, "cycle_time"):
             self.cycle_time = {}
         out_dict = {}
         to_do = [r]
         while to_do:
             current = to_do.pop()
-            cycle = self.get_successors(current)
-            _next = self.successor.get(cycle[-1], [])
+            cycle = custom_successor(current,time)
```
Comment (leoguignard) [inline src/LineageTree/lineageTree.py:2066]: This is a number of time points, not a cycle right??

**Review by leoguignard — COMMENTED** [review]
I really have a hard time understanding what it is you'd like to do.

Also it really looks like that your code works in your specific example but that there are many examples where it would break.

For example, what happens if a cell cycle is longer than your `time` parameter but the cell still has a successor?

You'll say that the cell cycle length is `time` while it's longer and then look at the daughter cells?

Few things make sense in your code ...

File: src/LineageTree/lineageTree.py (hunk 3 above)
Comment (BadPrograms) [reply src/LineageTree/lineageTree.py:2066]: So , lets say I have a lineage treacked up to 500 timepoint and another up to 400 timepoint, so to make a correct comparison I either have to trim the .mastodon or make a cutoff. I chose the latter. Also I do not think it breaks to be honest.

Comment (leoguignard) [reply src/LineageTree/lineageTree.py:2066]: Ok, let's discuss that more specifically in a meeting, I am not sure our views align on the best way to do that :D

Comment (BadPrograms) [reply src/LineageTree/lineageTree.py:2066]: Alright!

**Review by leoguignard — APPROVED** [review]

**Pull request merged**